### ISSUE TITLE
Oembedded Honeypoint

### ISSUE THREAD

**Issue by beardyunixer** [friendica/friendica#3543]
Somebody shared a link to a honeypot, see this post:

https://soc.beardyunixer.com/display/0c40ea672959416207bf24e053727771

As it was a naked link, we tried to oembed it, which got me flagged at spamhaus (relevant, but long winded part of the even longer winded spamhaus page quoted at the end), and in trouble with my host.

I'm not entirely sure what the fix for this looks like.  I'm even less convinced it's our problem - it's not exactly sophisticated on their end - but I can't see them fixing it either.  I'm not sure what to do..,we might need to check their blocklists before oembedding anything.  I know I'm not the only person with this problem - this probably explains the recent spike in "my mails aren't being sent" posts across the network - I'm just the first to figure out what happened.

It's a bit of an edge case, I know, but an important one - and innocent(ish) link can actually get accounts suspended.

> IP Address 212.24.102.104 is listed in the CBL. It shows signs of being infected with a spam sending trojan, malicious link or some other form of botnet.

> It was last detected at 2017-06-14 16:00 GMT (+/- 30 minutes), approximately 5 days, 21 hours ago.

> This IP is infected with, or is NATting for a machine infected with a Trojan.

> This was detected by observing this IP attempting to make contact to a sinkhole that is specifically designed to capture such attempts.

> This was detected by a TCP/IP connection from "212.24.102.104" on port "n/a" going to IP address "184.105.192.2" (the sinkhole) on port "80".

> The botnet command and control domain for this connection was "asdfasdfasdfasdf.com".

> Behind a NAT, you should be able to find the infected machine by looking for attempted connections to IP address "184.105.192.2" or host name "asdfasdfasdfasdf.com" on any port with a network sniffer such as wireshark. Equivalently, you can examine your DNS server or proxy server logs to references to "184.105.192.2" or "asdfasdfasdfasdf.com". See Advanced Techniques for more detail on how to use wireshark - ignore the references to port 25/SMTP traffic - the identifying activity is NOT on port 25.

> Please note that some of the above quoted information may be empty ("") or "na" or "-". In those cases, the feed has declined or is unable to give us that information. Hopefully enough information will be present to allow you to pinpoint the connections. If not, the destination ports to check are usually port 80, 8080, 443 or high ports (around 16000) outbound from your network. Most of these infections spray these connections in high volume, and they should stand out.

> This detection corresponds to a connection at 2017-06-14 16:20:40 (GMT - this timestamp is believed accurate to within one second).

**Comment by annando** [comment]
I got flagged as well ... So now I know why.

**Comment by beardyunixer** [comment]
I've edited the post now to remove the link now someone else has seen it and knows what's going on....but having slept on it, I don't see how we can even in principle fix this.

The blocklists don't publish the honeypots, they publish the IPs that have touched the honeypot.  We can't know if a URL is a honeypot until after someone has been flagged.

This time it was innocent, by someone who should know better, but innocent...but imagine someone posting this to a busy forum (if we ever get those).

Is there a "Can't fix" tag.  I feel like this should be open, but can't in principle be fixed.

**Comment by tobiasd** [comment]
I've created one for you

**Comment by dawnbreak** [comment]
Is this about that the honeypot got hit once, or that the oembed is proxying ever hit on the post through your server and therefore mass hitting the honeypot?

**Comment by beardyunixer** [comment]
We only hit it once (or possibly a few times - IIRC, there's some regression in there as 'retries') and then we cache it.  So possibly more than once, but not repeatedly.  The CBL documentation says the list isn't maintained by volume - hitting it once seems to be enough. 

It does say it looks for characteristics specific to open proxies.  I'm not sure if Friendica uses a proxy, I haven't delved into this bit for quite a while now...but I think it does, so maybe that's part of it.

One suspects you'll have the same problem at Hubzilla either way - even if it does check if you look like a proxy, sslify() will probably look like a proxy.

**Comment by AlfredSK** [comment]
I'm not listed in the SBL, PBL, XBL. Phew! :)
As far as I can tell spamhaus.org has a removal process. This doesn't solve the problem but may help in the short term.
I wonder how the big (proprietary) networks solve this. They use oembed too.

**Comment by beardyunixer** [comment]
Yeah, the removal process is easy.  It's not so much having your server marked as "spam" that's the problem - it's when your host watches the blacklists, and suspended your account for it.

Once I was suspended, they let me back in for 24 hours to find the problem.  It took me about 45 minutes (or about three hours - I'd already spent a couple of hours reading all the spamhaus doco before I started).  It was simple enough, but only because I'm familiar with the code.  Anybody else would have been screwed.

That's why I think the bug should be open, even though I really can't see how we can fix it.

**Comment by annando** [comment]
We could try to get in contact with spamhaus, so we could ask them about a solution to this.

**Comment by MrPetovan** [comment]
I've been flagged as well, and I removed myself from the CBL, but the CloudFlare page still appears, probably until I complete the CAPTCHA at least once, but how can I complete the CAPTCHA from my server? Would using Squid as a reverse proxy from by laptop browser work?
